tokio-threadpool/src/worker/entry.rs

```diff
@@ -22,6 +24,9 @@ pub(crate) struct WorkerEntry {
     // comments on that type.
     pub state: CachePadded<AtomicUsize>,
 
+    // The set of tasks registered in this worker entry.
+    owned_tasks: Mutex<HashSet<*const Task>>,
```

Comment:
It seems that the mutex isn't *really* required, it is just here to avoid additional unsafe code (which is a good goal). In practice, the mutex should not be contended, but I wonder what the perf impact of it would be on platforms that do not have good mutex implementations.

Did this change impact benchmarks that spawn many tasks?

Another option would be to avoid the mutex in `owned_tasks`, then have an `owned_tasks_shutdown` field with the mutex. Before the worker terminates, it moves the HashSet. Or, a channel between the worker and the shutdown trigger is used to send the hashset.

Reply:
The perf impact is minimal (see the comment below). How do you feel about switching to `parking_lot`, though? It has a very efficient implementation on all platforms.

Reply:
Well, avoiding the mutex entirely is doable. Do you think the strategy I mentioned above isn't worth it or too complicated?

Reply:
It is doable, but I think the complexity to gains ratio doesn't justify it. :) Maybe once/if it becomes a problem in the future...